packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Program.cs

```diff
@@ -0,0 +1,258 @@
+            var portingAssistantPortingConfiguration = System.Text.Json.JsonSerializer.Deserialize<PortingAssistantPortingConfiguration>(File.ReadAllText(config));
+            string metricsFolder = Path.Combine(userData, "telemetry-logs");
+            Directory.CreateDirectory(metricsFolder);
+            TelemetryConfiguration teleConfig = new TelemetryConfiguration{
+              InvokeUrl = portingAssistantPortingConfiguration.PortingAssistantMetrics["InvokeUrl"].ToString(),
+              Region = portingAssistantPortingConfiguration.PortingAssistantMetrics["Region"].ToString(),
+              ServiceName = portingAssistantPortingConfiguration.PortingAssistantMetrics["ServiceName"].ToString(),
+              Description = portingAssistantPortingConfiguration.PortingAssistantMetrics["Description"].ToString(),
+              LogsPath = metricsFolder,
+              LogFilePath = Path.Combine(metricsFolder, $"portingAssistant-telemetry-{DateTime.Today.ToString("yyyyMMdd")}.log"),
+              MetricsFilePath = Path.Combine(metricsFolder, $"portingAssistant-telemetry-{DateTime.Today.ToString("yyyyMMdd")}.metrics"),
+              Suffix = new List<string>(){".log", ".metrics"}
+            };
+            var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
+            string prefix = portingAssistantPortingConfiguration.PortingAssistantMetrics["Prefix"].ToString();
+            var client = new HttpClient();
+
+            // Create a timer and set an interval.
+            var logTimer = new System.Timers.Timer();
+            logTimer.Interval = Convert.ToDouble(portingAssistantPortingConfiguration.PortingAssistantMetrics["LogTimerInterval"].ToString());
+            // Hook up the Elapsed event for the timer. 
+            // logTimer.Elapsed += OnTimedEvent;
+            logTimer.Elapsed += (source, e) => OnTimedEvent(source, e, teleConfig, lastReadTokenFile, client, profile, prefix);
+
+            // Have the timer fire repeated events (true is the default)
+            logTimer.AutoReset = true;
+
+            // Start the timer
+            logTimer.Enabled = true;
+
+            // LogWatcher logWatcher = new LogWatcher(teleConfig, profile, prefix);
+            // logWatcher.Start();
+            _connection.Listen();
+        }
+
+      private class PortingAssistantPortingConfiguration
+        {
+            public PortingAssistantConfiguration PortingAssistantConfiguration { get; set; }
+            public Dictionary<string, object> PortingAssistantMetrics { get; set; }
+        }
+      
+      private static void OnTimedEvent(Object source, System.Timers.ElapsedEventArgs e, TelemetryConfiguration teleConfig, string lastReadTokenFile, HttpClient client, string profile, string prefix )
+        {
+          try {
+            // Get files in directory and filter based on Suffix
+            string[] fileEntries = Directory.GetFiles(teleConfig.LogsPath).Where(f => (
```

Comment:
Where do we add a check for existence of this directory? If the Path does not exist this would throw an exception, which might stop the Timed Event. If we do not validate the existence of this directory, we can either have a conditional block and continue, or handle the expected Exception without throwing an exception

Reply:
The try-catch block should handle the exception.